include/xtensor/xaxis_iterator.hpp

```diff
@@ -81,27 +75,42 @@ namespace xt
     template <class E>
     auto axis_begin(E&& e);
 
+    template <class E>
+    auto axis_begin(E&& e, typename std::decay_t<E>::size_type axis);
+
     template <class E>
     auto axis_end(E&& e);
 
+    template <class E>
+    auto axis_end(E&& e, typename std::decay_t<E>::size_type axis);
+
     /*********************************
      * xaxis_iterator implementation *
      *********************************/
 
-    template <class CT>
-    template <class T>
-    inline std::enable_if_t<std::is_pointer<T>::value, std::add_lvalue_reference_t<std::remove_pointer_t<T>>>
-    xaxis_iterator<CT>::deref(T val) const
-    {
-        return *val;
-    }
-
-    template <class CT>
-    template <class T>
-    inline std::enable_if_t<!std::is_pointer<T>::value, T>
-    xaxis_iterator<CT>::deref(T& val) const
-    {
-        return val;
+    namespace detail
+    {
+        template <class CT>
+        auto derive_xstrided_view(CT&& e, typename std::decay_t<CT>::size_type axis, typename std::decay_t<CT>::size_type offset)
+        {
+            using xexpression_type = std::decay_t<CT>;
+            using size_type = typename xexpression_type::size_type;
+            using shape_type = typename xexpression_type::shape_type;
+            using strides_type = typename xexpression_type::strides_type;
+
+            const auto& e_shape = e.shape();
+            shape_type shape(e_shape.size() - 1);
+            auto nxt = std::copy(e_shape.cbegin(), e_shape.cbegin() + axis, shape.begin());
+            std::copy(e_shape.cbegin() + axis + 1, e_shape.end(), nxt);
+
+            const auto& e_strides = e.strides();
+            strides_type strides(e_strides.size() - 1);
+            auto nxt_strides = std::copy(e_strides.cbegin(), e_strides.cbegin() + axis, strides.begin());
+            std::copy(e_strides.cbegin() + axis + 1, e_strides.end(), nxt_strides);
+
+            return strided_view(std::forward<CT>(e), std::forward<shape_type>(std::move(shape)),
+                std::forward<strides_type>(std::move(strides)), offset, e.layout());
```

Comment:
No need to call `std::forward` here, `std::move` is enough:
`return strided_view(std::forward<CT>(e), std::move(shape), std::move(strides), offset, e.layout());`

Reply:
Removed forward.